redbeams/src/main/resources/schema/app/20190501151942_CB-1174_database_server_config.sql

```diff
@@ -0,0 +1,44 @@
+-- CB-1174 database server config
+
+CREATE TABLE databaseserverconfig
+(
+   id                   BIGINT NOT NULL,
+   workspace_id         INT8,
+   name                 VARCHAR(255) NOT NULL DEFAULT 'name',
+   description          TEXT,
+   host                 VARCHAR(255),
+   port                 INT4,
+   databasevendor       VARCHAR(255) NOT NULL,
+   connectiondriver     VARCHAR(255) DEFAULT 'org.postgresql.Driver',
+   connectionusername   VARCHAR(255) NOT NULL,
+   connectionpassword   VARCHAR(255) NOT NULL,
+   creationdate         BIGINT,
+   resourcestatus       VARCHAR(255) NOT NULL DEFAULT 'USER_MANAGED',
+   connectorjarurl      VARCHAR(255),
+   archived             BOOLEAN DEFAULT FALSE,
+   deletionTimestamp    BIGINT DEFAULT -1,
+   PRIMARY KEY (id)
+);
+
+CREATE SEQUENCE databaseserverconfig_id_seq START WITH 1 INCREMENT BY 1 NO MINVALUE NO MAXVALUE CACHE 1;
+
+CREATE UNIQUE INDEX IF NOT EXISTS databaseserverconfig_id_idx ON databaseserverconfig(id);
+CREATE INDEX IF NOT EXISTS databaseserverconfig_name_idx ON databaseserverconfig(name);
+CREATE INDEX IF NOT EXISTS databaseserverconfig_workspace_id_idx ON databaseserverconfig(workspace_id);
+CREATE INDEX IF NOT EXISTS databaseserverconfig_workspace_id_name_idx ON databaseserverconfig(workspace_id, name);
+
```

Comment:
another constraint could also be needed if this is archivable:
ADD CONSTRAINT uk_databaseserverconfig_deletiondate UNIQUE (name, deletionTimestamp);

Reply:
Adding it. I had something like it commented out, but including workspace_id, so if workspaces do go away, it will end up exactly matching your suggestion.